File: 코드샘플/메타토이드래곤즈/유니티/System/Data/Datas/QuestData.cs
using Newtonsoft.Json.Linq;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
namespace SandboxNetwork
{
    public class QuestData : TableData<DBQuest_base>
    {
        static private QuestTable table = null;
        static public QuestData Get(string key)
        {
            if (table == null)
                table = TableManager.GetTable<QuestTable>();

            return table.Get(key);
        }

        static public QuestData Get(int key)
        {
            if (table == null)
                table = TableManager.GetTable<QuestTable>();

            return table.Get(key);
        }

        static public eQuestType GetQuestType(int key)
        {
            if (table == null)
                table = TableManager.GetTable<QuestTable>();

            var data = table.Get(key);
            if (data == null)
                return eQuestType.MAIN;
            return data.TYPE;
        }

        public int KEY { get { return Int(UNIQUE_KEY); } }
        public eQuestType TYPE => SBFunc.ConvertToQuestType(Data.TYPE);
        public eQuestGroup GROUP => (eQuestGroup)Data.GROUP;
        public string ICON => Data.ICON;
        public string _SUBJECT => Data._SUBJECT;
        public bool IS_START_POPUP => Data.IS_START_POPUP == 1;
        public int START_GROUP => Data.START_GROUP;
        public int CONDITION_GROUP => Data.CONDITION_GROUP;
        public int REWARD_ACCOUNT_EXP => Data.REWARD_ACCOUNT_EXP;
        public int REWARD_GOLD => Data.REWARD_GOLD;
        public int REWARD_ENERGY => Data.REWARD_ENERGY;
        public int REWARD_TICKET_ARENA => Data.REWARD_TICKET_PVP;
        public int REWARD_GROUP => Data.REWARD_GROUP;

        public int EVENT_KEY => Data.EVENT_KEY;

        private List<Asset> rewards = null;
        public List<Asset> REWARDS {
            get {
                if (rewards == null)
                {
                    rewards = new List<Asset>();
                    if (REWARD_ACCOUNT_EXP > 0)
                    {
                        rewards.Add(new Asset(eGoodType.NONE, 10000003, REWARD_ACCOUNT_EXP));
                    }

                    if (REWARD_GOLD > 0)
                    {
                        rewards.Add(new Asset(eGoodType.GOLD, 10000001, REWARD_GOLD));
                    }

                    if (REWARD_ENERGY > 0)
                    {
                        rewards.Add(new Asset(eGoodType.ENERGY, 10000002, REWARD_ENERGY));
                    }

                    if (REWARD_TICKET_ARENA > 0)
                    {
                        rewards.Add(new Asset(eGoodType.ARENA_TICKET, 10000007, REWARD_TICKET_ARENA));
                    }

                    if(REWARD_GROUP > 0)
                    {
                        var rewardItemList = ItemGroupData.Get(REWARD_GROUP);
                        if(rewardItemList != null && rewardItemList.Count > 0)
                        {
                            foreach(var rewardItem in rewardItemList)
                            {
                                rewards.Add(rewardItem.Reward);
                            }
                        }

                    }
                }
                return rewards; 
            } 
        }
	}
    public class QuestTriggerData : TableData<DBQuest_trigger_group>
    {
        static private QuestTriggerTable table = null;
        static public List<QuestTriggerData> Get(string key)
        {
            if (table == null)
                table = TableManager.GetTable<QuestTriggerTable>();

            if(int.TryParse(key, out int iKey))
                return table.GetByGroup(iKey);

            return null;
        }

        static public QuestTriggerData GetTrigger(int triggerID)
        {
            if (table == null)
                table = TableManager.GetTable<QuestTriggerTable>();

            return table.Get(triggerID);
        }

        public string KEY { get { return UNIQUE_KEY; } }
        public int TRIGGER_TYPE => Data.TRIGGER_TYPE;
        public int GROUP => Data.GROUP;
        public string TYPE => Data.TYPE;
        public string SUB_TYPE => Data.SUB_TYPE;
        public string TYPE_KEY => Data.TYPE_KEY;
        public int TYPE_KEY_VALUE => Data.TYPE_KEY_VALUE;
        public string _NOTIE => Data._NOTIE;
    }
}

File: 코드샘플/메타토이드래곤즈/코코스크리에이터/Scripts/Data/QuestData.ts
import { DataBase } from "./DataBase";

export class QuestData extends DataBase {
    protected qtype : number = 0
    public get QTYPE() { return this.qtype;}
    public set QTYPE(value : number) { this.qtype = value;}

    protected icon : string = ""
    public get ICON() { return this.icon;}
    public set ICON(value : string) { this.icon = value;}

    protected subject : number = 0
    public get SUBJECT() { return this.subject;}
    public set SUBJECT(value : number) { this.subject = value;}

    protected noti : number = 0
    public get NOTI() { return this.noti;}
    public set NOTI(value : number) { this.noti = value;}

    protected sgroup : number = 0
    public get START_GROUP() { return this.sgroup;}
    public set START_GROUP(value : number) { this.sgroup = value;}

    protected cgroup : number = 0
    public get CONDITION_GROUP() { return this.cgroup;}
    public set CONDITION_GROUP(value : number) { this.cgroup = value;}

    protected reward_exp : number = 0
    public get REWARD_ACCOUNT_EXP() { return this.reward_exp;}
    public set REWARD_ACCOUNT_EXP(value : number) { this.reward_exp = value;}

    protected reward_gold : number = 0
    public get REWARD_GOLD() { return this.reward_gold;}
    public set REWARD_GOLD(value : number) { this.reward_gold = value;}

    protected reward_energy : number = 0
    public get REWARD_ENERGY() { return this.reward_energy;}
    public set REWARD_ENERGY(value : number) { this.reward_energy = value;}

    protected reward_items_id : Array<number> = []
    public get REWARD_ITEMS_ID() { return this.reward_items_id;}
    public set REWARD_ITEMS_ID(value : Array<number>) { this.reward_items_id = value;}

    protected reward_items_value : Array<number> = []
    public get REWARD_ITEMS_VALUE() { return this.reward_items_value;}
    public set REWARD_ITEMS_VALUE(value : Array<number>) { this.reward_items_value = value;}
}

export class QuestTriggerData extends DataBase {
    protected key : number = 0
    public get KEY() { return this.key;}
    public set KEY(value : number) { this.key = value}

    protected trigger_type : number = 0
    public get TRIGGER_TYPE() { return this.trigger_type;}
    public set TRIGGER_TYPE(value : number) { this.trigger_type = value;}

    protected group : number = 0
    public get GROUP() { return this.group;}
    public set GROUP(value : number) { this.group = value;}

    protected type : string = ""
    public get TYPE() { return this.type;}
    public set TYPE(value : string) { this.type = value;}

    protected type_key : string|number;
    public get TYPE_KEY() { return this.type_key;}
    public set TYPE_KEY(value : string|number) { this.type_key = value;}

    protected type_key_value : number = 0
    public get TYPE_KEY_VALUE() { return this.type_key_value;}
    public set TYPE_KEY_VALUE(value : number) { this.type_key_value = value;}
}